pkg/operators/oci-handler/oci_test.go

```diff
@@ -0,0 +1,347 @@
+		},
+		{
+			name:            "missing dockerconfigjson key",
+			secretName:      "missingKey",
+			secretNamespace: "default",
+			secretType:      corev1.SecretTypeDockerConfigJson,
+			secretDataKey:   "someOtherKey",
+			secretContent:   "data",
+			expectError:     false,
+			expectedResult:  "",
+		},
+		{
+			name:            "empty secret content",
+			secretName:      "emptySecret",
+			secretNamespace: "default",
+			secretType:      corev1.SecretTypeDockerConfigJson,
+			secretDataKey:   corev1.DockerConfigJsonKey,
+			secretContent:   "",
+			expectError:     false,
+			expectedResult:  "",
+		},
+		{
+			name:            "valid special characters",
+			secretName:      "docker-pull",
+			secretNamespace: "custom-ns",
+			secretType:      corev1.SecretTypeDockerConfigJson,
+			secretDataKey:   corev1.DockerConfigJsonKey,
+			secretContent:   `$%{}!@#`,
+			expectError:     false,
+			expectedResult:  `$%{}!@#`,
+		},
+		{
+			name:             "secret in different namespace",
+			secretName:       "wrong-ns-secret",
+			secretNamespace:  "wrong-ns",
+			expectError:      true,
+			expectedErrorMsg: "getting secret \"wrong-ns-secret\"",
+		},
+		{
+			name:            "large content",
+			secretName:      "largeSecret",
+			secretNamespace: "default",
+			secretType:      corev1.SecretTypeDockerConfigJson,
+			secretDataKey:   corev1.DockerConfigJsonKey,
+			secretContent:   string(make([]byte, 1000)),
+			expectError:     false,
+			expectedResult:  string(make([]byte, 1000)),
+		},
+		{
+			name:            "nil data map",
+			secretName:      "nilDataSecret",
+			secretNamespace: "default",
+			secretType:      corev1.SecretTypeDockerConfigJson,
+			expectError:     false,
+			expectedResult:  "",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fcs := fake.NewSimpleClientset()
+
+			if test.secretType != "" {
+				secret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      test.secretName,
+						Namespace: test.secretNamespace,
+					},
+					Type: test.secretType,
+				}
+				if test.secretDataKey != "" {
+					secret.Data = map[string][]byte{
+						test.secretDataKey: []byte(test.secretContent),
+					}
+				}
+				_, err := fcs.CoreV1().Secrets(test.secretNamespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+				require.NoError(t, err, "setup: failed to create test secret")
+			}
+
+			result, err := getPullSecret(test.secretName, test.secretNamespace, fcs)
+
+			if test.expectError {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), test.expectedErrorMsg)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, test.expectedResult, string(result))
+			}
+		})
+	}
+}
+
+func TestConstructTempConfig(t *testing.T) {
+	tests := []struct {
+		name       string
+		input      string
+		wantErr    bool
+		wantConfig map[string]any
+		wantLength int
```

Comment:
I am not so sure about the naming. Its not the length we care about, but the type of annotation we get back from the function. Its either a `DataSource` annotation or a `Field` annotation